src/Rox/Core/XPack/Network/StateSender.php

```diff
@@ -169,17 +169,14 @@ private function _serializeCustomProperties()
         $mapped = array_map(function ($key) use ($allCustomProperties) {
             $value = (string) $allCustomProperties[$key];
 
-            if (!$this->_isCBP && $allCustomProperties[$key]->getType()->getExternalType() === "DateTime") {
-                return null;
-            }
-
             return json_decode($value, true);
         }, $keys);
 
-        return array_filter($mapped, function ($v) {
-            return $v !== null;
+        $mapped = array_filter($mapped, function ($v) {
+            return !(!$this->_isCBP && $v['externalType'] === "DateTime");
```

Comment:
are you sure this works?
value seems like a string. `$v['externalType']` will return what you want?

can you please add a test?

Reply:
btw - you can return it as is, without filtering if `_isCBP`
only if it's `!_isCBP` do the filtering

Reply:
@AsafRollout It's pretty much the same as before (during manual test), I just changed places; but I agree that this should be covered with unit tests, so I will add them